From: Nick Rosbrook <nick.rosbrook@canonical.com>
Date: Thu, 18 May 2023 11:55:47 -0400
Subject: test: drop /etc/default/locale handling from TEST-74-AUX-UTILS

This partially reverts commit bb59fdc1e3a7119f3680d309147020fce9bf67b5.
Only leave behind logic to create /etc/locale.conf if it's not already
there.
---
 test/units/testsuite-74.firstboot.sh | 34 +++++++++++++++-------------------
 1 file changed, 15 insertions(+), 19 deletions(-)

diff --git a/test/units/testsuite-74.firstboot.sh b/test/units/testsuite-74.firstboot.sh
index 92a6075..d9feae2 100755
--- a/test/units/testsuite-74.firstboot.sh
+++ b/test/units/testsuite-74.firstboot.sh
@@ -24,11 +24,7 @@ ROOT_HASHED_PASSWORD1='$6$foobarsalt$YbwdaATX6IsFxvWbY3QcZj2gB31R/LFRFrjlFrJtTTq
 # shellcheck disable=SC2016
 ROOT_HASHED_PASSWORD2='$6$foobarsalt$q.P2932zYMLbKnjFwIxPI8y3iuxeuJ2BgE372LcZMMnj3Gcg/9mJg2LPKUl.ha0TG/.fRNNnRQcLfzM0SNot3.'

-# Debian and Ubuntu use /etc/default/locale instead of /etc/locale.conf. Make
-# sure we use the appropriate path for locale configuration.
-LOCALE_PATH="/etc/locale.conf"
-[ -e "$LOCALE_PATH" ] || LOCALE_PATH="/etc/default/locale"
-[ -e "$LOCALE_PATH" ] || systemd-firstboot --locale=C.UTF-8
+[ -e /etc/locale.conf ] || systemd-firstboot --locale=C.UTF-8

 # Create a minimal root so we don't modify the testbed
 ROOT=test-root
@@ -37,14 +33,14 @@ mkdir -p "$ROOT/bin"
 touch "$ROOT/bin/fooshell" "$ROOT/bin/barshell"

 systemd-firstboot --root="$ROOT" --locale=foo
-grep -q "LANG=foo" "$ROOT$LOCALE_PATH"
-rm -fv "$ROOT$LOCALE_PATH"
+grep -q "LANG=foo" "$ROOT/etc/locale.conf"
+rm -fv "$ROOT/etc/locale.conf"
 systemd-firstboot --root="$ROOT" --locale-messages=foo
-grep -q "LC_MESSAGES=foo" "$ROOT$LOCALE_PATH"
-rm -fv "$ROOT$LOCALE_PATH"
+grep -q "LC_MESSAGES=foo" "$ROOT/etc/locale.conf"
+rm -fv "$ROOT/etc/locale.conf"
 systemd-firstboot --root="$ROOT" --locale=foo --locale-messages=bar
-grep -q "LANG=foo" "$ROOT$LOCALE_PATH"
-grep -q "LC_MESSAGES=bar" "$ROOT$LOCALE_PATH"
+grep -q "LANG=foo" "$ROOT/etc/locale.conf"
+grep -q "LC_MESSAGES=bar" "$ROOT/etc/locale.conf"

 systemd-firstboot --root="$ROOT" --keymap=foo
 grep -q "KEYMAP=foo" "$ROOT/etc/vconsole.conf"
@@ -88,8 +84,8 @@ systemd-firstboot --root="$ROOT" \
                   --root-password-hashed="$ROOT_HASHED_PASSWORD2" \
                   --root-shell=/bin/barshell \
                   --kernel-command-line="hello.world=0"
-grep -q "LANG=foo" "$ROOT$LOCALE_PATH"
-grep -q "LC_MESSAGES=bar" "$ROOT$LOCALE_PATH"
+grep -q "LANG=foo" "$ROOT/etc/locale.conf"
+grep -q "LC_MESSAGES=bar" "$ROOT/etc/locale.conf"
 grep -q "KEYMAP=foo" "$ROOT/etc/vconsole.conf"
 readlink "$ROOT/etc/localtime" | grep -q "Europe/Berlin$"
 grep -q "foobar" "$ROOT/etc/hostname"
@@ -109,8 +105,8 @@ systemd-firstboot --root="$ROOT" --force \
                   --root-password-hashed="$ROOT_HASHED_PASSWORD2" \
                   --root-shell=/bin/barshell \
                   --kernel-command-line="hello.world=0"
-grep -q "LANG=locale-overwrite" "$ROOT$LOCALE_PATH"
-grep -q "LC_MESSAGES=messages-overwrite" "$ROOT$LOCALE_PATH"
+grep -q "LANG=locale-overwrite" "$ROOT/etc/locale.conf"
+grep -q "LC_MESSAGES=messages-overwrite" "$ROOT/etc/locale.conf"
 grep -q "KEYMAP=keymap-overwrite" "$ROOT/etc/vconsole.conf"
 readlink "$ROOT/etc/localtime" | grep -q "/CET$"
 grep -q "hostname-overwrite" "$ROOT/etc/hostname"
@@ -124,7 +120,7 @@ rm -fr "$ROOT"
 mkdir "$ROOT"
 # Copy everything at once (--copy)
 systemd-firstboot --root="$ROOT" --copy
-diff $LOCALE_PATH "$ROOT$LOCALE_PATH"
+diff /etc/locale.conf "$ROOT/etc/locale.conf"
 diff <(awk -F: '/^root/ { print $7; }' /etc/passwd) <(awk -F: '/^root/ { print $7; }' "$ROOT/etc/passwd")
 diff <(awk -F: '/^root/ { print $2; }' /etc/shadow) <(awk -F: '/^root/ { print $2; }' "$ROOT/etc/shadow")
 [[ -e /etc/vconsole.conf ]] && diff /etc/vconsole.conf "$ROOT/etc/vconsole.conf"
@@ -133,7 +129,7 @@ rm -fr "$ROOT"
 mkdir "$ROOT"
 # Copy everything at once, but now by using separate switches
 systemd-firstboot --root="$ROOT" --copy-locale --copy-keymap --copy-timezone --copy-root-password --copy-root-shell
-diff $LOCALE_PATH "$ROOT$LOCALE_PATH"
+diff /etc/locale.conf "$ROOT/etc/locale.conf"
 diff <(awk -F: '/^root/ { print $7; }' /etc/passwd) <(awk -F: '/^root/ { print $7; }' "$ROOT/etc/passwd")
 diff <(awk -F: '/^root/ { print $2; }' /etc/shadow) <(awk -F: '/^root/ { print $2; }' "$ROOT/etc/shadow")
 [[ -e /etc/vconsole.conf ]] && diff /etc/vconsole.conf "$ROOT/etc/vconsole.conf"
@@ -148,8 +144,8 @@ set +o pipefail
 # We can do only limited testing here, since it's all an interactive stuff,
 # so --prompt and --prompt-root-password are skipped on purpose
 echo -ne "\nfoo\nbar\n" | systemd-firstboot --root="$ROOT" --prompt-locale
-grep -q "LANG=foo" "$ROOT$LOCALE_PATH"
-grep -q "LC_MESSAGES=bar" "$ROOT$LOCALE_PATH"
+grep -q "LANG=foo" "$ROOT/etc/locale.conf"
+grep -q "LC_MESSAGES=bar" "$ROOT/etc/locale.conf"
 # systemd-firstboot in prompt-keymap mode requires keymaps to be installed so
 # it can present them as a list to the user. As Debian does not ship/provide
 # compatible keymaps (from the kbd package), skip this test if the keymaps are
